contrib/AutoGrouper.lua

```diff
@@ -0,0 +1,155 @@
+end
+
+local function InRange(test, low, high) --tests if test value is within range of low and high (inclusive)
+	if test >= low and test <= high then
+		return true
+	else
+		return false
+	end
+end
+
+local function CompTime(first, second) --compares the timestamps and returns true if first was taken before second
+	first_time = first.exif_datetime_taken
+	if string.match(first_time, '[0-9]') == nil then first_time = '9999:99:99 99:99:99' end
+	first_time = tonumber(string.gsub(first_time, '[^0-9]*',''))
+	second_time = second.exif_datetime_taken
+	if string.match(second_time, '[0-9]') == nil then second_time = '9999:99:99 99:99:99' end
+	second_time = tonumber(string.gsub(second_time, '[^0-9]*',''))
+	return first_time < second_time
+end
+
+local function SeperateTime(str) --seperates the timestamp into individual components for used with OS.time operations
+	local cleaned = string.gsub(str, '[^%d]',':')
+	cleaned = string.gsub(cleaned, '::*',':')  --YYYY:MM:DD:hh:mm:ss
+	local year = string.sub(cleaned,1,4)
+	local month = string.sub(cleaned,6,7)
+	local day = string.sub(cleaned,9,10)
+	local hour = string.sub(cleaned,12,13)
+	local min = string.sub(cleaned,15,16)
+	local sec = string.sub(cleaned,18,19)
+	return {year = year, month = month, day = day, hour = hour, min = min, sec = sec}
+end
+
+local function GetTimeDiff(curr_image, prev_image) --returns the time difference (in sec.) from current image and the previous image
+	local curr_time = SeperateTime(curr_image.exif_datetime_taken)
+	local prev_time = SeperateTime(prev_image.exif_datetime_taken)
+	return os.time(curr_time)-os.time(prev_time)
+end
+
+local function main(on_collection)
+	local images = {}
+	if on_collection then 
+		local col_images = dt.collection
+		for i,image in ipairs(col_images) do --copy images to a standard table, table.sort barfs on type dt_lua_singleton_image_collection
+			table.insert(images,i,image)
+		end
+	else
+		images = dt.gui.selection()
+	end
+	dt.preferences.write(mod, 'active_gap', 'integer', GUI.gap.value)
+	if #images < 2 then return end
+	table.sort(images, function(first, second) return CompTime(first,second) end)  --sort images by timestamp
+	
+	for i, image in ipairs(images) do
+		if i == 1 then 
+			prev_image = image
+		elseif string.match(image.exif_datetime_taken, '[%d]') ~= nil then --make sure current image has a timestamp, if so check if it is within the user specified gap value and add to group
+			local curr_image = image
+			if GetTimeDiff(curr_image, prev_image) <= GUI.gap.value then
+				images[i]:group_with(images[i-1])
+			end
+			prev_image = curr_image
+		end
+	end
+end
+
+-- GUI --
+GUI = {
+	gap = 			{},
+	selected = 		{},
+	collection = 	{}
+}
+temp = dt.preferences.read(mod, 'active_gap', 'integer')
+if not InRange(temp, 1, 3600) then temp = 3 end
+GUI.gap = dt.new_widget('slider'){
+	label = _('group gap [sec.]'),
+	tooltip = _('minimum gap, in seconds, between groups'),
+	soft_min = 1,
+	soft_max = 30,

```

Comment:
Is this high enough?  Should it maybe be a minute or two, or five?  I tried it on a baseball game I shot and I ended up at 200 seconds.  I tried 300, but that was a little too much.  I also tried a 2 basketball games shot back to back.  With a 30 second gap I got 45 groups.  I think the right gap for that is probably 60 seconds, which would group between timeouts. Lastly I tried a senior picture session I did.  The 30 second grouping was just about right and the grouping was pretty close to what I did manually when I processed them.

Reply:
The hard max is 3600 so the user can enter up to a full 1hr gap. I don't really know what would make a good soft max. 

My only thought was to avoid a frustration I have with some other sliders where the adjustments I am trying to make are difficult due to being stuck on the far end of the slider range. This happens to me frequently when using the right-click method for adjusting white balance, It's hard to make fine tune adjustments when lowering WB because I'm already so close to the left-hand boundary. 

That being said, I don't have a strong preference here, so if you all feel a higher soft max would be more usable by the majority of people then let's make it whatever you think is best.

Reply:
I'm not sure either.  I toyed with the idea of 3 sliders, hours, minutes, and seconds, but I'm not sure that's any better.  How about we make it 60 seconds, and then see if the users provide any feedback.  Or, we can leave it as is and see if the user's have feedback.  Up to you.  Let me know.

Reply:
60 sounds good, a nice even 1 minute. I'll do that and take care of these other notes you've made when I get some time to tinker with these again. Glad you got some use out of it though.

Reply:
I made it 60, I also increased the hard max all the way up to a full day. I saw no drawbacks to having a very large hard max.